src/platform/ESP32/NetworkCommissioningDriver.h

```diff
@@ -0,0 +1,129 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <esp_wifi.h>
+#include <platform/NetworkCommissioning.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace NetworkCommissioning {
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
+class ESPScanResponseIterator : public Iterator<WiFiScanResponse>
+{
+public:
+    ESPScanResponseIterator(size_t size, wifi_ap_record_t * scanResults) : mSize(size), mpscanResults(scanResults) {}
+    size_t Count() override { return mSize; }
+    bool Next(WiFiScanResponse & item) override
+    {
+        if (mSize == 0 || iternum >= mSize)
+        {
+            return false;
+        }
+
+        item.security = mpscanResults[iternum].authmode;
+        item.ssidLen  = strnlen((const char *) mpscanResults[iternum].ssid, chip::DeviceLayer::Internal::kMaxWiFiSSIDLength);
+        item.channel  = mpscanResults[iternum].primary;
+        item.wiFiBand = chip::DeviceLayer::NetworkCommissioning::WiFiBand::k2g4;
+        item.rssi     = mpscanResults[iternum].rssi;
+        memcpy(item.ssid, mpscanResults[iternum].ssid, item.ssidLen);
+        memcpy(item.bssid, mpscanResults[iternum].bssid, 6);
+
+        iternum++;
+        return true;
+    }
+    void Release() override {}
+
+private:
+    size_t mSize;
+    wifi_ap_record_t * mpscanResults;
+    uint8_t iternum = 0;
+};
+
+class ESPWiFiDriver final : public WiFiDriver
+{
+public:
+    class WiFiNetworkIterator final : public NetworkIterator
+    {
+    public:
+        WiFiNetworkIterator(ESPWiFiDriver * aDriver) : driver(aDriver) {}
+        size_t Count() override;
+        bool Next(Network & item) override;
+        void Release() override { delete this; }
+        ~WiFiNetworkIterator() = default;
+
+    private:
+        ESPWiFiDriver * driver;
```

Comment:
nit: generally we have members of classes named `mMemberName`, so this could be `mDriver` and `mExhausted`.

Reply:
Changed the members name.